cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Instruction.qll

```diff
@@ -368,6 +369,12 @@ class Instruction extends Construction::TInstruction {
     )
   }
 
+  int getLineRank() {
+    this = rank[result](Instruction instr | instr.getAST().getFile() = getAST().getFile() and instr.getAST().getLocation().getStartLine() = getAST().getLocation().getStartLine() | instr
```

Comment:
This pattern of `x.getFile() = y.getFile() and x.getStartLine() = y.getStartLine()` requires some luck in optimization. Please check the tuple counts for bulges.